test/FunctionalTests/HealthChecks.Uris/UriHealthCheckTests.cs

```diff
@@ -267,5 +269,91 @@ public async Task be_healthy_if_request_sucess_and_timeout_is_configured()
             response.StatusCode
                     .Should().Be(HttpStatusCode.OK);
         }
+
+        [Fact]
+        public async Task be_healthy_if_request_success_and_response_data_is_configured()
+        {
+            string url = "https://httpbin.org/status/200";
+            var uri = new Uri(url);
+
+            var webHostBuilder = new WebHostBuilder()
+              .UseStartup<DefaultStartup>()
+              .ConfigureServices(services =>
+              {
+                  services.AddHealthChecks()
+                   .AddUrlGroup(opt =>
+                   {
+                       opt.AddUri(uri);
+
+                       opt.AddResponseData(ResponseData.Body);
+                       opt.AddResponseData(ResponseData.HttpVerb);
+                       opt.AddResponseData(ResponseData.Reason);
+                       opt.AddResponseData(ResponseData.Status);
+                       opt.AddResponseData(ResponseData.Url);
+
+                   }, tags: new string[] { "uris" });
+              })
+              .Configure(app =>
+              {
+                  app.UseHealthChecks("/health", new HealthCheckOptions()
+                  {
+                      Predicate = r => r.Tags.Contains("uris"),
+                      ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+                  });
+              });
+
+            var server = new TestServer(webHostBuilder);

```

Comment:
The whole TestServer and WebHostBuilder code could be refactored into a function to reduce code duplication.

Reply:
The same code appears in `be_unhealthy_if_request_success_and_response_data_is_configured` unit test